test/unit/query_builder_test.rb

```diff
@@ -257,6 +257,28 @@ def test_find_with_args
     assert_requested find_stub, times: 1
   end
 
+  def test_find_with_blank_args
+    blank_params = [nil, '', {}]
+
+    with_altered_config(Article, :raise_on_blank_find_param => true) do
+      blank_params.each do |blank_param|
+        assert_raises JsonApiClient::Errors::NotFound do
+          Article.find(blank_param)
+        end
+      end
+    end
+
+    # `.find('')` hits the INDEX URL with trailing slash, the others without one..
+    collection_stub = stub_request(:get, %r{\Ahttp://example.com/articles/?\z})
+        .to_return(headers: {content_type: "application/vnd.api+json"}, body: { data: [] }.to_json)
+
+    blank_params.each do |blank_param|
+      Article.find(blank_param)
+    end
+
+    assert_requested collection_stub, times: 3
+  end
```

Comment:
This looks good! Can we add a `BlankRaisingResource.all` test to make sure that's not broken?

Reply:
@gaorlov Sure things, already in place!